src/renderer/Renderer.ts

```diff
@@ -0,0 +1,171 @@
+        const fieldOfView = Math.PI / 4;
+        const aspect = width / height;
+        const zNear = 1;
+        const zFar = 1000;
+        mat4.perspective(this.projectionMatrix, fieldOfView, aspect, zNear, zFar);
+
+        // Set up drawing commands
+        const regl: REGL.Regl = REGL(canvas3D);
+
+        this.clearAll = () => {
+            this.ctx2D.clearRect(0, 0, this.width, this.height);
+            regl.clear({
+                color: [0, 0, 0, 1],
+                depth: 1
+            });
+        };
+
+        this.clearDepth = () =>
+            regl.clear({
+                depth: 1
+            });
+
+        this.drawObject = drawObject(regl);
+        this.drawAxes = drawAxes(regl);
+    }
+
+    public draw(objects: RenderObject[], debug: boolean = false) {
+        this.clearAll();
+
+        objects.forEach((o: RenderObject) =>
+            this.drawObject({
+                model: o.transform,
+                cameraTransform: this.cameraTransform,
+                projectionMatrix: this.projectionMatrix,
+                positions: o.vertices,
+                normals: o.normals,
+                colors: o.colors,
+                indices: o.indices
+            })
+        );
+
+        if (debug) {
+            this.drawCrosshairs();
+        }
+    }
+
+    private drawCrosshairs() {
+        this.clearDepth();
+
+        const [zero, x, y, z] = [[0, 0, 0], [1, 0, 0], [0, 1, 0], [0, 0, 1]].map(
+            (point: number[]) => {
+                // Initially treat these as vectors (w = 0) instead of points (where w would be 1)
+                // so that only the direction changes, and they are not translated from the origin
+                const vector = vec4.fromValues(point[0], point[1], point[2], 0);
+
+                // Bring them into camera space
+                vec4.transformMat4(vector, vector, this.cameraTransform);
+
+                // Scale them and place them in the lower left corner of the screen
+                vec4.scale(vector, vector, Math.min(this.width, this.height) * 0.05);
+                vec4.add(vector, vector, [this.width * -0.25, this.height * -0.25, -500, 1]);
+
+                // Project them into 2D coordinates
+                vec4.transformMat4(vector, vector, this.projectionMatrix);
+
+                return vector;
+            }
+        );
+
+        const colorX = [1, 0, 0];
+        const colorY = [0, 1, 0];
+        const colorZ = [0, 0, 1];
+
+        this.drawAxes({
+            positions: [zero, x, zero, y, zero, z],
+            colors: [colorX, colorX, colorY, colorY, colorZ, colorZ],
+            count: 6
+        });
+
+        // Use the 2D projected points to draw text labels for the axes. To convert the GL 3D point
+        // to a point where each element is in [0, 1], we use:
+        //   point2D = (point3D / point3D.w + 1) / 2
+        // ...and then multiply by the width/height of the screen.
+        this.ctx2D.fillStyle = '#FF0000';
```

Comment:
This is a nit, but could we create variables for these hex colour strings - just so it's more apparent what the colour is? I know we're using simple values (i.e., this is red, the next is blue, etc.)